parameters.py

```diff
@@ -3,80 +3,95 @@
         file.write(content)
-
     return filepath
 
-def readODTPConfigYAML(configfile):
-    with open(configfile, 'r') as file:
-        data = yaml.safe_load(file)
-     
-    return data
-
-def obtainAllPlaceholders(filepath):
-    template = readTemplate(filepath)
 
+def __obtainAllPlaceholders(filepath):
+    """returns all placeholders in a template file
+    :param filepath: input file path
+    :return: all placeholders
+    """
+    template = __readTemplate(filepath)
     pattern = r'\{\{([^{}]+)\}\}'
 
     # Find all placeholders in the template
     placeholders = re.findall(pattern, template)
 
     return placeholders
 
-     
-
-if __name__ == "__main__":
-
-    print("PREPARATION OF CONFIG FILE")
-
-    if len(sys.argv) < 3:
-        print("Please provide a filepath and output filepaht as arguments.")
-        sys.exit(1)  # Exit the script with an error code
-
-    filepath = sys.argv[1]
-    outputfilepath = sys.argv[2]
-
-    odtpConfig = readODTPConfigYAML("/odtp/odtp-config/odtp.yml")
-    #argumentsdefault = odtpConfig["arguments"]
-    #Take the default from here
 
-    # Implementing simple load of parameters by now. Not based in the odtp config file, as both pipelines contains different arguments name. 
-    placeholders = obtainAllPlaceholders(filepath)
+def __process_file(filepath, outputfilepath):
+    """
+    a file is processed:
+    :param filepath: file to process
+    :param outputfilepath: output path
+    """
+    placeholders = __obtainAllPlaceholders(filepath)
 
     # We expect those values to be in environment
     placeholderValue = []
+    placeholders_not_found = []
     for placeholder in placeholders:
         try:
             placeholderValue.append(os.environ[placeholder])
         except Exception as e:
             print(f"{placeholder} not found")
             print(e)
-
+            placeholders_not_found.append(placeholder)
+            placeholderValue.append("NotFound")
 
     # Replacing template
-    template = readTemplate(filepath)
-    templateFilled = replaceListParameters(template, placeholders, placeholderValue)
-    _ = saveConfigFile(templateFilled, outputfilepath)
+    template = __readTemplate(filepath)
+    templateFilled = __replaceListParameters(template, placeholders, placeholderValue)
+    outputfilepath = __saveConfigFile(templateFilled, outputfilepath)
+    print(f"Outputfile available: {outputfilepath}.")
+    if placeholders_not_found:
+        print(f"Some placeholders could not be filled: check your environment "
+              f"variables to resolve the errors.")
 
-    print("PREPARATION OF CONFIG DONE")
 
+if __name__ == "__main__":
+    # This python script is called from external: it reads a config file and
+    # output file path: it replaces all placeholder values marked by {{ .. }}
+    # with environment variables of the same name: so {{ HELLO }} is replaced
+    # by the value in $HELLO
+    print("PREPARATION OF CONFIG FILE")
```

Comment:
Maybe we can use here `logging.info`?

Reply:
@caviri Logging needs to be configured to work. This can be done. But for now I would like to keep it simple.